docs/shared/core-tutorial/05-auto-detect-dependencies.md

```diff
@@ -0,0 +1,104 @@
+# Core Nx Tutorial - Step 5: Automatically Detect Dependencies
+
+Manually telling Nx about the dependencies between your projects is helpful, but as your repo grows it becomes difficult for a person to keep track of all the dependencies that are introduced. If you miss a dependency, the guarantees of the affected command are voided.
+
+Luckily, Nx core can automatically detect dependencies that are created in `.ts` or `.js` files.
```

Comment:
I'd add a point saying that it can do it for other technologies with plugins

The way it works is that we analyze js/ts source and package.json files. 

You could say that you can turn of ts/js analysis and only rely on package.json files if you rely on say yarn workspaces to run your monorepo cause then the project graph will match